### PR TITLE
traffic-graphs, optimize retrieval of data every x seconds hopefully reducing 'spikes' and lowering cpu usage on pfSense

### PR DESCRIPTION
**Pull request by PiBa-NL**
traffic-graphs, optimize retrieval of data every x seconds hopefully reducing 'spikes' and lowering cpu usage on pfSense

in my test vm request time recorded by Chrome developer network tab the time drops from 100ms to 5ms per request

'spikes' like reported here: https://forum.pfsense.org/index.php?topic=126010.msg695920

### CODE REVIEW

**Comment by PiBa-NL** [comment]
@jdillard , rebased for easier merging/testing.. hoping it will get reviewed/merged before conflicts happen again.

**Comment by PiBa-NL** [comment]
Hi @jdillard could you take a look at this pullrequest? It has been waiting for some time now, and would be a pity to see merge conflicts appear. Thanks in advance.

**Comment by jdillard** [comment]
The title doesn't seem to match most of the code changes. I also see some authentication handling and adding real interface names. Are you referring to the changes made to `src/usr/local/www/js/traffic-graphs.js`?

**Comment by PiBa-NL** [comment]
@jdillard the title covers the changes for the most part.. :) To elaborate a bit though, basically there are two changes here that make up for the improved nginx response speed.
Also these changes were made before the 'divide bandwidth by time' patch 3607 was created before which sometimes graphs showed huge spikes because traffic-graph requests took several seconds to process when pfSense was 'bussy'. This pullrequest does optimize data retrieval and lower cpu. As for reducing the spikes that part is not really relevant anymore.

- using this new auth_check.inc avoids expensive inclusion of 'priv.inc' and possibly other includes while still providing a authentication check that no unauthorized requests are handled.
- the call to  get_real_interface($if); is also somewhat expensive and looking up the real interface once on page load once avoids needing to get the real interface every second again and again.
The changes in traffic-graph.js are used to make the this change work, but doesn't really change affect performance by itself. It does break functionality though if used without the other changes..

Both these changes account for cutting down the response time for the webgui requests made and overall cpu usage for nginx/php on pfSense while having the graphs open.

Also removed some old/unused/commented code..

Anyhow i hope this helps understand what i have done/intended. Lemme know if something isn't clear thanks.

**Comment by PiBa-NL** [comment]
@jdillard, ping ?

**Comment by PiBa-NL** [comment]
@jdillard can you please perform this review? thanks.

**Comment by PiBa-NL** [comment]
@jdillard i moved the auth_check code to a different pullrequest #3728 , making this one easier to review.

**Comment by PiBa-NL** [comment]
@jdillard ping, pong..?

**Comment by PiBa-NL** [comment]
rebased again.. @jdillard @sbeaver-netgate @rbgarga , can someone please pick it up?

**Comment by PiBa-NL** [comment]
@jdillard please i beg you.. can you approve / pull this request?

File: src/usr/local/www/widgets/widgets/traffic_graphs.widget.php [hunk 1]
```diff
@@ -251,102 +253,22 @@
 	}
 	echo json_encode($iflist);
 ?>;
-
+a
```
Comment (jdillard) [inline src/usr/local/www/widgets/widgets/traffic_graphs.widget.php]: This appears to be a typo

**Comment by PiBa-NL** [comment]
@jdillard, no clue where that extra "a" came from (probably wrote it myself while testing 'something'..) removed it now.

**Pull request merged**